### PR TITLE
🆙 Update for Node 16 support

### PR DESCRIPTION
**Pull request by stoe**
This pull request includes several updates to the codebase, focusing on modernizing dependencies, improving configurations, and refactoring code for better maintainability and readability. The most important changes include updates to ESLint and Prettier configurations, workflow adjustments, package updates, and refactoring of the codebase to use ES modules.

### Configuration and Dependency Updates:
* [`.eslintrc.json`](diffhunk://#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fL3-R4): Updated `ecmaVersion` to 2020 and `sourceType` to `module`. Adjusted ESLint rules for `camelcase` and added `no-console` rule. [[1]](diffhunk://#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fL3-R4) [[2]](diffhunk://#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fL14-R22)
* [`.github/workflows/codeql.yml`](diffhunk://#diff-12783128521e452af0cfac94b99b8d250413c516ec71fe6d97dbea666ff7ba27R18-R37): Added environment variables and updated actions to specific commit hashes.
* [`.github/workflows/test.yml`](diffhunk://#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88L26-R26): Updated Node.js version matrix to use `20.x`.
* [`action.yml`](diffhunk://#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6L23-R23): Updated Node.js version to `20` for the action runtime.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R3-R14): Added `type: "module"`, updated dependencies and devDependencies, and modified scripts for better compatibility and performance. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R3-R14) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L19-R59)

### Code Migration to ES Modules:
* [`src/index.js`](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L1-R25): Refactored to use ES module syntax and updated imports accordingly. [[1]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L1-R25) [[2]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L38-R43)
* [`src/report.js`](diffhunk://#diff-535233921fc6c419b8396f283b26c74a63db6e33f86f46a66197e50f7b7b098aL1-R3): Converted to ES module and updated import statements. [[1]](diffhunk://#diff-535233921fc6c419b8396f283b26c74a63db6e33f86f46a66197e50f7b7b098aL1-R3) [[2]](diffhunk://#diff-535233921fc6c419b8396f283b26c74a63db6e33f86f46a66197e50f7b7b098aL41-R46) [[3]](diffhunk://#diff-535233921fc6c419b8396f283b26c74a63db6e33f86f46a66197e50f7b7b098aL287-L288)

### Testing and Linting:
* [`tests/index.t.js`](diffhunk://#diff-4a029227247a72bbede870405584683c297f5bffb37deb80a0b0a616396eea85L1-L72): Removed the old CommonJS-based test file.
* [`tests/report.t.js`](diffhunk://#diff-842b1d6bbc74a637e91d5ab8fcbf5ec52254ec53bdb1134f2163961560535d0dL1-R10): Updated to use ES module syntax and adjusted mocks accordingly. [[1]](diffhunk://#diff-842b1d6bbc74a637e91d5ab8fcbf5ec52254ec53bdb1134f2163961560535d0dL1-R10) [[2]](diffhunk://#diff-842b1d6bbc74a637e91d5ab8fcbf5ec52254ec53bdb1134f2163961560535d0dL47-R50)

### Other Changes:
* [`.prettierrc.cjs`](diffhunk://#diff-c490b8b502570136250b231654638d2ce43379b01978f96cd4c7f9ab662ad7fcR1-R2): Added ESLint directive and required Prettier configuration from `eslint-plugin-github`.
* Removed `prettier.config.js` as it is now included in `.prettierrc.cjs`.

### CODE REVIEW

**Comment by stoe** [comment]
Next step after merge: Create new release!

File: action.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ outputs:
     description: 'Report head SHA'
 
 runs:
-  using: 'node12'
+  using: 'node16'
```
Comment (gclhub) [inline action.yml:23]:
```suggestion
  using: 'node20'
```

**Review by gclhub — CHANGES_REQUESTED** [review]
@stoe I saw that you already have a PR to a newer nodeJS version in this repo, so I added a couple of suggestions to upgrade to node v20 in response to the prompt from @stebje to upgrade the ActionsDesk repositories.

File: package.json [hunk 2]
```diff
@@ -16,46 +21,41 @@
...
-  },
-  "publishConfig": {
-    "registry": "https://npm.pkg.github.com/"
-  },
+  "repository": "github:ActionsDesk/ghec-invitations-report-action",
   "engines": {
-    "node": ">=14",
-    "npm": ">=7"
+    "node": ">=16",
+    "npm": ">=8"
```
Comment (stoe) [inline package.json:27]:
```suggestion
    "node": ">=20",
    "npm": ">=10"
```

**Review by gclhub — APPROVED** [review]
LGTM :shipit:

**Pull request merged**